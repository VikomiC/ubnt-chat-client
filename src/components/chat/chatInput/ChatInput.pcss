 .chatInputContainer {
    bottom: 25px;
    left: 25px;
    right: 25px;
    position: fixed;
}
