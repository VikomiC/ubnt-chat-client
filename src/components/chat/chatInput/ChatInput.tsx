import * as React from "react";

import TextField, { TextFieldProps } from "@material-ui/core/TextField";

import * as styles from "./ChatInput.pcss";

interface ChatInputState {
  chatMessage: string;
}

interface ChatInputProps {
  onChatMessageEnter: (chatmessage: string) => void;
}

export class ChatInput extends React.Component<ChatInputProps, ChatInputState> {
  public state: ChatInputState = {
    chatMessage: "",
  };

  public render() {
    const { chatMessage } = this.state;

    const helperText = `${chatMessage.length} / 140`;

    const textFieldProps: TextFieldProps = {
      error: chatMessage.length > 140,
      fullWidth: true,
      helperText,
      label: "Type your message",
      onChange: this.onChatMessageChange,
      onKeyPress: (ev: any) => {
        if (ev.key === "Enter") {
          ev.preventDefault();
          this.onEnter();
        }
      },
      value: chatMessage,
    };

    return (
      <div className={styles.chatInputContainer}>
        <TextField {...textFieldProps} />
      </div>
    );
  }

  private onChatMessageChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ chatMessage: e.currentTarget.value });
  }

  private onEnter = () => {
    const { chatMessage } = this.state;
    const { onChatMessageEnter } = this.props;
    if (chatMessage && chatMessage.length <= 140) {
      onChatMessageEnter(chatMessage);
      this.setState({ chatMessage: "" });
    }
  }
}
