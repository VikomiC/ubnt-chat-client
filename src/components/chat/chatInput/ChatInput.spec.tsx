import { mount } from "enzyme";
import * as React from "react";

import { ChatInput } from "./ChatInput";

describe("components/chat/chatInput/ChatInput", () => {
    it("should match snapshot for ChatInput", () => {
        const component = mount(<ChatInput onChatMessageEnter={() => {/*noop*/}} />);
        expect(component).toMatchSnapshot();
    });
});
