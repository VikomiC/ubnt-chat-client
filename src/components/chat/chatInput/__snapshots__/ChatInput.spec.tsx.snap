// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/chat/chatInput/ChatInput should match snapshot for ChatInput 1`] = `
<ChatInput
  onChatMessageEnter={[Function]}
>
  <div>
    <TextField
      error={false}
      fullWidth={true}
      helperText="0 / 140"
      label="Type your message"
      onChange={[Function]}
      onKeyPress={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    >
      <WithStyles(FormControl)
        error={false}
        fullWidth={true}
        onKeyPress={[Function]}
        required={false}
        variant="standard"
      >
        <FormControl
          classes={
            Object {
              "fullWidth": "MuiFormControl-fullWidth-4",
              "marginDense": "MuiFormControl-marginDense-3",
              "marginNormal": "MuiFormControl-marginNormal-2",
              "root": "MuiFormControl-root-1",
            }
          }
          component="div"
          disabled={false}
          error={false}
          fullWidth={true}
          margin="none"
          onKeyPress={[Function]}
          required={false}
          variant="standard"
        >
          <div
            className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
            onKeyPress={[Function]}
          >
            <WithStyles(WithFormControlContext(InputLabel))>
              <WithFormControlContext(InputLabel)
                classes={
                  Object {
                    "animated": "MuiInputLabel-animated-13",
                    "disabled": "MuiInputLabel-disabled-7",
                    "error": "MuiInputLabel-error-8",
                    "filled": "MuiInputLabel-filled-14",
                    "focused": "MuiInputLabel-focused-6",
                    "formControl": "MuiInputLabel-formControl-10",
                    "marginDense": "MuiInputLabel-marginDense-11",
                    "outlined": "MuiInputLabel-outlined-15",
                    "required": "MuiInputLabel-required-9",
                    "root": "MuiInputLabel-root-5",
                    "shrink": "MuiInputLabel-shrink-12",
                  }
                }
              >
                <InputLabel
                  classes={
                    Object {
                      "animated": "MuiInputLabel-animated-13",
                      "disabled": "MuiInputLabel-disabled-7",
                      "error": "MuiInputLabel-error-8",
                      "filled": "MuiInputLabel-filled-14",
                      "focused": "MuiInputLabel-focused-6",
                      "formControl": "MuiInputLabel-formControl-10",
                      "marginDense": "MuiInputLabel-marginDense-11",
                      "outlined": "MuiInputLabel-outlined-15",
                      "required": "MuiInputLabel-required-9",
                      "root": "MuiInputLabel-root-5",
                      "shrink": "MuiInputLabel-shrink-12",
                    }
                  }
                  disableAnimation={false}
                  muiFormControl={
                    Object {
                      "adornedStart": false,
                      "disabled": false,
                      "error": false,
                      "filled": false,
                      "focused": false,
                      "margin": "none",
                      "onBlur": [Function],
                      "onEmpty": [Function],
                      "onFilled": [Function],
                      "onFocus": [Function],
                      "required": false,
                      "variant": "standard",
                    }
                  }
                >
                  <WithStyles(WithFormControlContext(FormLabel))
                    className="MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                    classes={
                      Object {
                        "disabled": "MuiInputLabel-disabled-7",
                        "error": "MuiInputLabel-error-8",
                        "focused": "MuiInputLabel-focused-6",
                        "required": "MuiInputLabel-required-9",
                      }
                    }
                    data-shrink={false}
                  >
                    <WithFormControlContext(FormLabel)
                      className="MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                      classes={
                        Object {
                          "asterisk": "MuiFormLabel-asterisk-22",
                          "disabled": "MuiFormLabel-disabled-18 MuiInputLabel-disabled-7",
                          "error": "MuiFormLabel-error-19 MuiInputLabel-error-8",
                          "filled": "MuiFormLabel-filled-20",
                          "focused": "MuiFormLabel-focused-17 MuiInputLabel-focused-6",
                          "required": "MuiFormLabel-required-21 MuiInputLabel-required-9",
                          "root": "MuiFormLabel-root-16",
                        }
                      }
                      data-shrink={false}
                    >
                      <FormLabel
                        className="MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                        classes={
                          Object {
                            "asterisk": "MuiFormLabel-asterisk-22",
                            "disabled": "MuiFormLabel-disabled-18 MuiInputLabel-disabled-7",
                            "error": "MuiFormLabel-error-19 MuiInputLabel-error-8",
                            "filled": "MuiFormLabel-filled-20",
                            "focused": "MuiFormLabel-focused-17 MuiInputLabel-focused-6",
                            "required": "MuiFormLabel-required-21 MuiInputLabel-required-9",
                            "root": "MuiFormLabel-root-16",
                          }
                        }
                        component="label"
                        data-shrink={false}
                        muiFormControl={
                          Object {
                            "adornedStart": false,
                            "disabled": false,
                            "error": false,
                            "filled": false,
                            "focused": false,
                            "margin": "none",
                            "onBlur": [Function],
                            "onEmpty": [Function],
                            "onFilled": [Function],
                            "onFocus": [Function],
                            "required": false,
                            "variant": "standard",
                          }
                        }
                      >
                        <label
                          className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
                          data-shrink={false}
                        >
                          Type your message
                        </label>
                      </FormLabel>
                    </WithFormControlContext(FormLabel)>
                  </WithStyles(WithFormControlContext(FormLabel))>
                </InputLabel>
              </WithFormControlContext(InputLabel)>
            </WithStyles(WithFormControlContext(InputLabel))>
            <WithStyles(Input)
              fullWidth={true}
              onChange={[Function]}
              value=""
            >
              <Input
                classes={
                  Object {
                    "disabled": "MuiInput-disabled-26",
                    "error": "MuiInput-error-28",
                    "focused": "MuiInput-focused-25",
                    "formControl": "MuiInput-formControl-24",
                    "fullWidth": "MuiInput-fullWidth-30",
                    "input": "MuiInput-input-31",
                    "inputMarginDense": "MuiInput-inputMarginDense-32",
                    "inputMultiline": "MuiInput-inputMultiline-33",
                    "inputType": "MuiInput-inputType-34",
                    "inputTypeSearch": "MuiInput-inputTypeSearch-35",
                    "multiline": "MuiInput-multiline-29",
                    "root": "MuiInput-root-23",
                    "underline": "MuiInput-underline-27",
                  }
                }
                fullWidth={true}
                onChange={[Function]}
                value=""
              >
                <WithStyles(WithFormControlContext(InputBase))
                  classes={
                    Object {
                      "disabled": "MuiInput-disabled-26",
                      "error": "MuiInput-error-28",
                      "focused": "MuiInput-focused-25",
                      "formControl": "MuiInput-formControl-24",
                      "fullWidth": "MuiInput-fullWidth-30",
                      "input": "MuiInput-input-31",
                      "inputMarginDense": "MuiInput-inputMarginDense-32",
                      "inputMultiline": "MuiInput-inputMultiline-33",
                      "inputType": "MuiInput-inputType-34",
                      "inputTypeSearch": "MuiInput-inputTypeSearch-35",
                      "multiline": "MuiInput-multiline-29",
                      "root": "MuiInput-root-23 MuiInput-underline-27",
                      "underline": null,
                    }
                  }
                  fullWidth={true}
                  inputComponent="input"
                  multiline={false}
                  onChange={[Function]}
                  type="text"
                  value=""
                >
                  <WithFormControlContext(InputBase)
                    classes={
                      Object {
                        "adornedEnd": "MuiInputBase-adornedEnd-41",
                        "adornedStart": "MuiInputBase-adornedStart-40",
                        "disabled": "MuiInputBase-disabled-39 MuiInput-disabled-26",
                        "error": "MuiInputBase-error-42 MuiInput-error-28",
                        "focused": "MuiInputBase-focused-38 MuiInput-focused-25",
                        "formControl": "MuiInputBase-formControl-37 MuiInput-formControl-24",
                        "fullWidth": "MuiInputBase-fullWidth-45 MuiInput-fullWidth-30",
                        "input": "MuiInputBase-input-46 MuiInput-input-31",
                        "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-52",
                        "inputAdornedStart": "MuiInputBase-inputAdornedStart-51",
                        "inputMarginDense": "MuiInputBase-inputMarginDense-47 MuiInput-inputMarginDense-32",
                        "inputMultiline": "MuiInputBase-inputMultiline-48 MuiInput-inputMultiline-33",
                        "inputType": "MuiInputBase-inputType-49 MuiInput-inputType-34",
                        "inputTypeSearch": "MuiInputBase-inputTypeSearch-50 MuiInput-inputTypeSearch-35",
                        "marginDense": "MuiInputBase-marginDense-43",
                        "multiline": "MuiInputBase-multiline-44 MuiInput-multiline-29",
                        "root": "MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27",
                      }
                    }
                    fullWidth={true}
                    inputComponent="input"
                    multiline={false}
                    onChange={[Function]}
                    type="text"
                    value=""
                  >
                    <InputBase
                      classes={
                        Object {
                          "adornedEnd": "MuiInputBase-adornedEnd-41",
                          "adornedStart": "MuiInputBase-adornedStart-40",
                          "disabled": "MuiInputBase-disabled-39 MuiInput-disabled-26",
                          "error": "MuiInputBase-error-42 MuiInput-error-28",
                          "focused": "MuiInputBase-focused-38 MuiInput-focused-25",
                          "formControl": "MuiInputBase-formControl-37 MuiInput-formControl-24",
                          "fullWidth": "MuiInputBase-fullWidth-45 MuiInput-fullWidth-30",
                          "input": "MuiInputBase-input-46 MuiInput-input-31",
                          "inputAdornedEnd": "MuiInputBase-inputAdornedEnd-52",
                          "inputAdornedStart": "MuiInputBase-inputAdornedStart-51",
                          "inputMarginDense": "MuiInputBase-inputMarginDense-47 MuiInput-inputMarginDense-32",
                          "inputMultiline": "MuiInputBase-inputMultiline-48 MuiInput-inputMultiline-33",
                          "inputType": "MuiInputBase-inputType-49 MuiInput-inputType-34",
                          "inputTypeSearch": "MuiInputBase-inputTypeSearch-50 MuiInput-inputTypeSearch-35",
                          "marginDense": "MuiInputBase-marginDense-43",
                          "multiline": "MuiInputBase-multiline-44 MuiInput-multiline-29",
                          "root": "MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27",
                        }
                      }
                      fullWidth={true}
                      inputComponent="input"
                      muiFormControl={
                        Object {
                          "adornedStart": false,
                          "disabled": false,
                          "error": false,
                          "filled": false,
                          "focused": false,
                          "margin": "none",
                          "onBlur": [Function],
                          "onEmpty": [Function],
                          "onFilled": [Function],
                          "onFocus": [Function],
                          "required": false,
                          "variant": "standard",
                        }
                      }
                      multiline={false}
                      onChange={[Function]}
                      type="text"
                      value=""
                    >
                      <div
                        className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          className="MuiInputBase-input-46 MuiInput-input-31"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          type="text"
                          value=""
                        />
                      </div>
                    </InputBase>
                  </WithFormControlContext(InputBase)>
                </WithStyles(WithFormControlContext(InputBase))>
              </Input>
            </WithStyles(Input)>
            <WithStyles(WithFormControlContext(FormHelperText))>
              <WithFormControlContext(FormHelperText)
                classes={
                  Object {
                    "contained": "MuiFormHelperText-contained-57",
                    "disabled": "MuiFormHelperText-disabled-55",
                    "error": "MuiFormHelperText-error-54",
                    "filled": "MuiFormHelperText-filled-59",
                    "focused": "MuiFormHelperText-focused-58",
                    "marginDense": "MuiFormHelperText-marginDense-56",
                    "required": "MuiFormHelperText-required-60",
                    "root": "MuiFormHelperText-root-53",
                  }
                }
              >
                <FormHelperText
                  classes={
                    Object {
                      "contained": "MuiFormHelperText-contained-57",
                      "disabled": "MuiFormHelperText-disabled-55",
                      "error": "MuiFormHelperText-error-54",
                      "filled": "MuiFormHelperText-filled-59",
                      "focused": "MuiFormHelperText-focused-58",
                      "marginDense": "MuiFormHelperText-marginDense-56",
                      "required": "MuiFormHelperText-required-60",
                      "root": "MuiFormHelperText-root-53",
                    }
                  }
                  component="p"
                  muiFormControl={
                    Object {
                      "adornedStart": false,
                      "disabled": false,
                      "error": false,
                      "filled": false,
                      "focused": false,
                      "margin": "none",
                      "onBlur": [Function],
                      "onEmpty": [Function],
                      "onFilled": [Function],
                      "onFocus": [Function],
                      "required": false,
                      "variant": "standard",
                    }
                  }
                >
                  <p
                    className="MuiFormHelperText-root-53"
                  >
                    0 / 140
                  </p>
                </FormHelperText>
              </WithFormControlContext(FormHelperText)>
            </WithStyles(WithFormControlContext(FormHelperText))>
          </div>
        </FormControl>
      </WithStyles(FormControl)>
    </TextField>
  </div>
</ChatInput>
`;
